jax_healpy/sphtfunc.py

```diff
@@ -328,33 +640,1069 @@ def map2alm(
+
+
+@partial(jax.jit, static_argnames=['lmax', 'mmax', 'lmax_out', 'nspec', 'healpy_ordering'])
+def alm2cl(
+    alms1: ArrayLike | list[ArrayLike],
+    alms2: ArrayLike | list[ArrayLike] | None = None,
+    lmax: int | None = None,
+    mmax: int | None = None,
+    lmax_out: int | None = None,
+    nspec: int | None = None,
+    healpy_ordering: bool = True,
+) -> ArrayLike | tuple[ArrayLike, ...]:
+    """Compute power spectra from spherical harmonic coefficients.
+
+    Calculates auto-spectra and cross-spectra from one or more sets of alm coefficients.
+    For n input alm arrays, produces n(n+1)/2 spectra containing all auto- and cross-spectra.
+
+    Parameters
+    ----------
+    alms1 : array-like or list of arrays
+        Spherical harmonic coefficients. Can be:
+        - Single array for one alm
+        - List/tuple of arrays for multiple alms (produces n(n+1)/2 spectra)
+    alms2 : array-like or list of arrays, optional
+        Second set of alm coefficients for cross-spectra. If None, computes auto-spectra.
+    lmax : int, optional
+        Maximum l value of input alm. If None, inferred from array size.
+    mmax : int, optional
+        Maximum m value of input alm. Default: lmax
+    lmax_out : int, optional
+        Maximum l value in output spectra. If None, uses lmax from input.
+        Note: Currently must equal lmax if provided.
+    nspec : int, optional
+        Number of spectra to return. If None, returns all.
+    healpy_ordering : bool, optional
+        If True, input alms use healpy 1D format. If False, uses s2fft 2D format.
+        Default: True
+
+    Returns
+    -------
+    cl : array-like or tuple of arrays
+        Power spectrum/spectra. Shape (lmax+1,) for single spectrum.
+        For multiple alms: tuple of n(n+1)/2 spectra in order: (11, 12, 22, 13, 23, 33, ...)
+
+    Notes
+    -----
+    The power spectrum is computed as:
+        C_l = (1 / (2*l + 1)) * sum_m |a_lm|^2  (auto-spectrum)
+        C_l = (1 / (2*l + 1)) * sum_m a_lm * conj(a'_lm)  (cross-spectrum)
+
+    For multiple input alms, the function computes all unique auto- and cross-spectra.
+    The ordering follows healpy convention: spectra are returned in diagonal-major order.
+
+    Examples
+    --------
+    Compute auto-spectrum from a single alm:
+
+    >>> import jax.numpy as jnp
+    >>> import jax_healpy as jhp
+    >>> alm = jnp.array([...])  # Your alm coefficients
+    >>> cl = jhp.alm2cl(alm)
+
+    Compute cross-spectrum between two alms:
+
+    >>> cl_cross = jhp.alm2cl(alm1, alm2)
+
+    Compute all spectra from multiple alms:
+
+    >>> alms = [alm1, alm2, alm3]
+    >>> cl_tuple = jhp.alm2cl(alms)  # Returns (cl11, cl12, cl22, cl13, cl23, cl33)
+    """
+    # Check if lmax_out is provided and different from lmax
+    if lmax_out is not None and lmax is not None and lmax_out != lmax:
+        raise ValueError(f'lmax_out ({lmax_out}) must equal lmax ({lmax}) if both provided')
+
```

Comment:
```suggestion
    if lmax_out < 0:
        raise ValueError(f'lmax_out ({lmax_out}) must equal positive')
```
If you are ok with the suggestion below about `lmax_out` (and adding a little test, although maybe not necessary)